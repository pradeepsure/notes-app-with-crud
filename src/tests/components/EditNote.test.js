import EditNote from '../../components/EditNote';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

let wrapper, notes, updateNote;
beforeEach(() => {    
    notes= [{
        "noteTitle": "Flipkart Sale",
        "noteDescription": "Electronics sale upto 50% off",
        "id": 1
      }]
    updateNote = jest.fn();
    wrapper = shallow(
        <EditNote  
        notes={notes}
        match={{params: {id: 1}, isExact: true, path: "", url: ""}}
        updateNote={updateNote}/>
    );
});


test('Render EditNote component correctly', () => {        
    expect(wrapper).toMatchSnapshot();
});