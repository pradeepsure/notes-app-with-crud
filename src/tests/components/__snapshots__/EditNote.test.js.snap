// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render EditNote component correctly 1`] = `
<WithStyles(Dialog)
  aria-labelledby="Edit note form"
  onClose={[Function]}
  open={true}
>
  <WithStyles(DialogTitle)
    id="Edit note form"
  >
    Edit Note
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="note title"
      label="Note Title"
      margin="dense"
      onChange={[Function]}
      required={false}
      select={false}
      type="text"
      value="Flipkart Sale"
      variant="standard"
    />
    <TextField
      fullWidth={true}
      id="note description"
      label="Note Description"
      margin="dense"
      onChange={[Function]}
      required={false}
      select={false}
      type="text"
      value="Electronics sale upto 50% off"
      variant="standard"
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Update
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
